objection/commands/android/hooking.py

```diff
@@ -356,3 +358,95 @@ def search_methods(args: list) -> None:
+    target = None
+    for i in range(len(args)):
+        if args[i] == '--json':
+            target = i + 1
+        i = i + 1
+
+    if target is None:
+        return None
+    elif target < len(args):
+        return args[target]
+    else:
+        click.secho('Please specify a target file', bold=True)
+
+    return None
+
+
+
+def _should_print_only_classes(args: list) -> bool:
+   return '--only-classes' in args
+
+
+def enumerate(args: list) -> None:
+    """
+        Enumerates the current Android application for classes and methods.
+
+        :param query:
+        :return:
+    """
+    if len(clean_argument_flags(args)) <= 0:
+        click.secho('Usage: android hooking enumerate \'<class>!<method>\''
+                    '<optional overload> '
+                    '(optional: --dump-args) '
+                    '(optional: --dump-backtrace) '
+                    '(optional: --dump-return)'
+                    '(optional: --json <filename>)'
+                    '(optional: --only-classes)'
+                    '(optional: --quiet)', bold=True)
+        return
+
+    shouldDumpJSON = _should_dump_json(args)
+    shouldPrintOnlyClasses = _should_print_only_classes(args)
+    shouldWatchArgs = _should_dump_args(args)
+    shouldWatchRet= _should_dump_return_value(args)
+    shouldBeQuiet = _should_be_quiet(args)
+    shouldBacktrace = _should_dump_backtrace(args)
+    overload_filter = args[1].replace(' ', '') if (len(args) > 1 and '--' not in args[1]) else None
+
+    api = state_connection.get_api()
+    results = api.android_hooking_enumerate(args[0])
+    # Only get overloads if this flag is specified, otherwise just enumerating can be kind of slow
+    if shouldDumpJSON:
+        for result in results:
+            for _class in result['classes']:
+                _class['overloads'] = api.android_hooking_get_class_methods_overloads(_class['name'], _class['methods'])
+
+    if shouldWatchArgs or shouldWatchRet or shouldBacktrace:
+        for result in results:
+            for _class in result['classes']:
+                classname = _class['name']
+                methods = _class['methods']
+                for method in methods:
+                    fullyQualifiedMethod = f'{classname}.{method}'
+                    api.android_hooking_watch_method(fullyQualifiedMethod,
+                                                     overload_filter,
+                                                     shouldWatchArgs,
+                                                     shouldBacktrace,
+                                                     shouldWatchRet)
+
+
+    if not shouldBeQuiet:
+        for result in results:
+            for _class in result['classes']:
+                print(_class['name'])
+                if not shouldPrintOnlyClasses:
+                    for method in _class['methods']:
+                        print(f'\t{method}')
+
+
+    targetFile = shouldDumpJSON
```

Comment:
This is an odd assignment, and I think an example of where the function return/responsibility is confusing based on the name.

Reply:
I have replaced this with:

```python
    if should_dump_json:
        target_file = _get_flag_value('--json', args)
        if target_file:
            results_json['data'] = results
            with open(target_file, 'w') as fd:
                fd.write(json.dumps(results_json))
                click.secho(f'JSON dumped to {should_dump_json}', bold=True)
```

Where the new `_get_flag` function looks like:

```python
def _get_flag_value(flag:str, args: list) -> Optional[str]:
    target = None
    for i in range(len(args)):
        if args[i] == flag:
            target = i + 1

    if target is None:
        return None
    elif target < len(args):
        return args[target]
    else:
        click.secho(f'Could not find specified value for {flag}', bold=True)
        return None
```

Leaving this as unresolved to see if your happy first :)